src/contracts/base/BaseParaSwapAdapter.sol

```diff
@@ -120,4 +120,25 @@ abstract contract BaseParaSwapAdapter is Ownable, IBaseParaSwapAdapter {
     POOL.withdraw(reserve, aTokenBalanceDiff, address(this));
     return aTokenBalanceDiff;
   }
+
+   /**
+   * @dev Renews the asset allowance in case the current allowance is below a given threshold
+   * @param asset The address of the asset
+   * @param minAmount The minimum required allowance to the Aave Pool
+   */
+  function _conditionalRenewAllowance(address asset, uint256 minAmount) internal {
+    uint256 allowance = IERC20(asset).allowance(address(this), address(POOL));
+    if (allowance < minAmount) {
+      _renewAllowance(asset);
+    }
+  }
+
+    /**
+   * @notice Renews the asset allowance to the Aave Pool
+   * @param reserve The address of the asset
+   */
+  function _renewAllowance(address reserve) internal {
```

Comment:
are we re-using this function somewhere else apart from `_conditionalRenewAllowance`?